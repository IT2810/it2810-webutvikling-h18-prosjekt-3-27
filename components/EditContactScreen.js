import React, { Component } from "react";
import {
  StyleSheet,
  View,
  Text,
  Button
} from "react-native";

export default class EditContactScreen extends Component {
  state = {
    name: "",
    number: "",
    key: 0,
  };

  handleDeletePress(){
    console.log("hola")
    if (this.state.name.trim().length === 0) {
      // invalid name
      return;
    }
    // retrieve deleteContact callback function
    const deleteContact = this.props.navigation.getParam("deleteContact");
    // contact that we want to delete
    const oldContact = {
      key: this.state.key.trim(),
    };
    deleteContact(oldContact);
    this.props.navigation.navigate("Contacts");
  }

  render() {
    console.log(this.state.name);
    console.log(this.state.key);
    return (
      <View>
        <Text>
          {this.state.name},
          {this.state.number}
        </Text>
        <Button
          color='pink'
          onPress={this.handleDeletePress}
          title={"delete"}
        />
      </View>
    )
  }

  componentDidMount() {
    const c = this.props.navigation.getParam("contact");
    console.log(c);
    this.setState({number: c.number});
    this.setState({name: c.name});
    this.setState({key: c.key});
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "stretch",
    backgroundColor: "white",
  },
});

