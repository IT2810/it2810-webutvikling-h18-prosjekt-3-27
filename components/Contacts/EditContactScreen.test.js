import React from 'react';
import EditContactScreen from './EditContactScreen';
import ShallowRenderer from 'react-test-renderer/shallow';


it("renders correctly", () => {
  const testContact = {
    name: 'mama',
    number: '1234',
    key: '1',
    color: 'green'
  };
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<EditContactScreen contact={testContact}/>);
  expect(tree).toMatchSnapshot();
});