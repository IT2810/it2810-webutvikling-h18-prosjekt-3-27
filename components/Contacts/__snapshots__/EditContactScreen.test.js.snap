// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "fontSize": 40,
        }
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
          "fontSize": 30,
          "margin": 20,
        }
      }
    >
      
    </Text>
  </Component>
  <Component
    style={
      Object {
        "bottom": 0,
        "marginBottom": 10,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Button
      color="green"
      onPress={[Function]}
      title="delete"
    />
  </Component>
</Component>
`;
